it/controllers/AtsMergePageControllerItSpec.scala

```diff
@@ -0,0 +1,319 @@
+
+      request.getQueryString(Globals.TAXS_AGENT_TOKEN_ID).isDefined mustBe true
+    }
+
+    "return an OK response when call to backend is successful but no data is found for either atsList or payeData" in {
+
+      server.stubFor(
+        get(urlEqualTo(backendUrlSa))
+          .willReturn(aResponse().withStatus(NOT_FOUND))
+      )
+
+      server.stubFor(
+        get(urlEqualTo(backendUrlPaye))
+          .willReturn(aResponse().withStatus(NOT_FOUND))
+      )
+
+      val request = FakeRequest(GET, url)
+
+      val result = route(fakeApplication(), request)
+
+      result.map(status) mustBe Some(OK)
+
+      request.getQueryString(Globals.TAXS_USER_TYPE_QUERY_PARAMETER) mustBe Some("PORTAL")
+
+      request.getQueryString(Globals.TAXS_AGENT_TOKEN_ID).isDefined mustBe true
+    }
+
+    List(
+      BAD_REQUEST,
+      IM_A_TEAPOT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { httpResponse =>
+      s"return an INTERNAL_SERVER_ERROR when the call to backend to retrieve atsList data throws a $httpResponse" in {
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlSa))
+            .willReturn(aResponse().withStatus(httpResponse))
+        )
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlPaye))
+            .willReturn(ok(FileHelper.loadFile("./it/resources/payeData.json")))
+        )
+
+        val request = FakeRequest(GET, url)
+
+        val result = route(fakeApplication(), request)
+
+        result.map(status) mustBe Some(INTERNAL_SERVER_ERROR)
+      }
+    }
+
+    List(
+      BAD_REQUEST,
+      IM_A_TEAPOT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { httpResponse =>
+      s"return an INTERNAL_SERVER_ERROR when the call to backend to retrieve payeData throws a $httpResponse" in {
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlSa))
+            .willReturn(ok(FileHelper.loadFile("./it/resources/atsList.json")))
+        )
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlPaye))
+            .willReturn(aResponse().withStatus(httpResponse))
+        )
+
+        val request = FakeRequest(GET, url)
+
+        val result = route(fakeApplication(), request)
+
+        result.map(status) mustBe Some(INTERNAL_SERVER_ERROR)
+      }
+    }
+
+    List(
+      BAD_REQUEST,
+      IM_A_TEAPOT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { httpResponse =>
+      s"return an INTERNAL_SERVER_ERROR when the call to backend to retrieve both atsList data and payeData throws a $httpResponse" in {
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlSa))
+            .willReturn(aResponse().withStatus(httpResponse))
+        )
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlPaye))
+            .willReturn(aResponse().withStatus(httpResponse))
+        )
+
+        val request = FakeRequest(GET, url)
```

Comment:
Not sure if this would be overkill but do we need to have tests for the empty token error scenarios as all these error cases are using the url with the token in

Reply:
@pascal22p ?

Reply:
No need. This ticket was to unblock an other ticket and this is about the empty token so it will get one.